lib/ts/superTokens.tsx

```diff
@@ -108,52 +103,48 @@ export default class SuperTokens {
         });
     }
 
+    private async initMultitenancyWithDynamicLoginMethods(config: SuperTokensConfig) {
+        const multitenancyRecipe =
+            config.recipeList.find(({ recipeID }) => recipeID === Multitenancy.RECIPE_ID) ?? Multitenancy.init({});
+        const appInfo = normaliseInputAppInfoOrThrowError(config.appInfo);
+        const enableDebugLogs = config.enableDebugLogs === true;
+
+        // initialize auth-react and web-js multitenancy recipes
+        const multitenancyInstance = multitenancyRecipe.authReact(appInfo, enableDebugLogs) as Multitenancy;
+        multitenancyRecipe.webJS(appInfo, config.clientType, enableDebugLogs);
+
+        const tenantID = multitenancyInstance.config.getTenantID?.();
+        const tenantMethods = await Multitenancy.getDynamicLoginMethods({ tenantId: tenantID });
+        const hasIntersection = hasIntersectingRecipes(tenantMethods, SuperTokens.getInstanceOrThrow().recipeList);
+
+        if (hasIntersection === false) {
+            throw new Error("Initialized recipes have no overlap with core recipes");
+        }
+
+        SuperTokens.uiController.emit("LoginMethodsLoaded");
```

Comment:
This doesn't feel like it belongs in `superTokens.tsx`. Couldn't we do this in the multitenancy recipe init? (or start it there).

Reply:
You can add it to the post init callbacks.